dicom-mr-classifier.py

```diff
@@ -528,31 +528,32 @@ def dicom_classify(zip_file_path, outbase, timezone, config=None):
         except:
             log.warning('Could not parse PatientAge, droppping.')
             pass
-    if hasattr(dcm, "PatientName") and dcm.get("PatientName").given_name:
-        # If the first name or last name field has a space-separated string, and one or the other field is not
-        # present, then we assume that the operator put both first and last names in that one field. We then
-        # parse that field to populate first and last name.
-        metadata["session"]["subject"]["firstname"] = str(
-            format_string(dcm.get("PatientName").given_name)
-        )
-        if not dcm.get("PatientName").family_name:
-            name = format_string(dcm.get("PatientName").given_name.split(" "))
-            if len(name) == 2:
-                first = name[0]
-                last = name[1]
-                metadata["session"]["subject"]["lastname"] = str(last)
-                metadata["session"]["subject"]["firstname"] = str(first)
-    if hasattr(dcm, "PatientName") and dcm.get("PatientName").family_name:
-        metadata["session"]["subject"]["lastname"] = str(
-            format_string(dcm.get("PatientName").family_name)
-        )
-        if not dcm.get("PatientName").given_name:
-            name = format_string(dcm.get("PatientName").family_name.split(" "))
-            if len(name) == 2:
-                first = name[0]
-                last = name[1]
-                metadata["session"]["subject"]["lastname"] = str(last)
-                metadata["session"]["subject"]["firstname"] = str(first)
+    if hasattr(dcm, "PatientName") and isinstance(dcm.get('PatientName'),pydicom.valuerep.PersonName):
+        if hasattr(dcm, "PatientName") and dcm.get("PatientName").given_name:
```

Comment:
You could leave out the `hasattr(dcm, "PatientName")` (from here and L546) check as you've already checked on 531.

Reply:
Done